lib/fluent/plugin/in_tail.rb

```diff
@@ -267,6 +268,9 @@ def shutdown
       @shutdown_start_time = Fluent::Clock.now
       # during shutdown phase, don't close io. It should be done in close after all threads are stopped. See close.
       stop_watchers(existence_path, immediate: true, remove_watcher: false)
+      @tails_rotate_wait.keys.each do |tw|
+        detach_watcher(tw, @tails_rotate_wait[tw][:ino])
+      end
```

Comment:
> during shutdown phase, don't close io. It should be done in close after all threads are stopped. See close.

I'm concerned about this point.

`stop_watchers()` with `immediate: true` calls `detach_watcher()` with `close_io: false`.

https://github.com/fluent/fluentd/blob/175d8669b8444505bd5168524b07d6a3133dca7e/lib/fluent/plugin/in_tail.rb#L503-L505

It seems that we should not close the io handler here.
We should do it in `close_watcher_handles()`, separately.

https://github.com/fluent/fluentd/blob/175d8669b8444505bd5168524b07d6a3133dca7e/lib/fluent/plugin/in_tail.rb#L275-L279

https://github.com/fluent/fluentd/blob/175d8669b8444505bd5168524b07d6a3133dca7e/lib/fluent/plugin/in_tail.rb#L512-L519

I'm concerned about these comments as well, but I think they are old.

https://github.com/fluent/fluentd/blob/175d8669b8444505bd5168524b07d6a3133dca7e/lib/fluent/plugin/in_tail.rb#L573-L577

At that time, `in_tail` used `Coolio::Loop`, but it is not used now.

* #348

I'm not sure if we really need to pay attention to the timing of the IO handler close...
At this point, I think it would be better to unify the timing of the process.
(Closing IO hanlders should be done in `close_watcher_handles()` when stopping.)

Reply:
> I'm not sure if we really need to pay attention to the timing of the IO handler close...

It may be because `io_handler` does not stop immediately.

https://github.com/fluent/fluentd/blob/175d8669b8444505bd5168524b07d6a3133dca7e/lib/fluent/plugin/in_tail.rb#L844-L850